File: LC461_HammingDistance/bitWise.py
'''
The Hamming distance between two integers is the number of positions at which the corresponding bits are different.
Given two integers x and y, return the Hamming distance between them.
'''


class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        diff = x ^ y
        result = 0
        while diff:
            temp = diff & 1
            result += temp
            diff >>= 1
        return result

File: LC461_HammingDistance/bitWise.js
/**
 * @param {number} x
 * @param {number} y
 * @return {number}
 */
 var hammingDistance = function(x, y) {
    // Bit Wise
    // Time Complexity: O(logx+logy)
    // Space Complexity: O(1)
    diff = x ^ y
    let result = 0
    while(diff){
        result += diff & 1
        diff >>>= 1
    }
    return result
};